### ISSUE TITLE
Open file with fuzzy search on workspace

### ISSUE THREAD

**Issue by Mistobaan** [Microsoft/vscode#9501]
I can't find a way to open a file under the current directory using visual studio code.
Ideally it would be like `CMD+P` but instead of opening only the recent ones, it will fuzzy search any file within the directory. 
I searched among the key bindings commands but did not find anything useful. Am I missing something ?

**Comment by kieferrm** [comment]
`Cmd+p` and then type what you are looking for.

<img width="459" alt="screen shot 2016-07-20 at 11 27 07 am" src="https://cloud.githubusercontent.com/assets/4674940/16998144/09f46350-4e6d-11e6-8040-e151b5479a96.png">

**Comment by Mistobaan** [comment]
my bad, it was so slow that I thought it did not work. Why is that slow ? any way to improve ?

**Comment by heybenchen** [comment]
@Mistobaan I would check to see if your project includes any large, generated directories like `tmp` and add them to the blacklist in your VSCode search settings.

**Comment by kieferrm** [comment]
@Mistobaan Cmd+P should be fast. When you invoke it the very first time it might take a few seconds more, but it should still be acceptable. Our test workspace is the Chromium workspace with more than 240000 files.  Since it seems to be slow for you, do you mind providing more info:
- What OS are you using?
- How many files does the workspace have that you're using?
- Are any folders network drives? 
- Anything else that comes to mind?
Thanks!

**Comment by aseiden** [comment]
@kieferrm I'm having a similar issue with my fuzzy search, so I'll let you know what's going on on my machine.

**OS:** macOS Sierra Version 10.12.5
**Files Count:** ~140k (but this includes several dirs I've added to `search.exclude` in my user settings.  I've ignored `node_modules`, `bower_components`, and a project specific `tmp` and `dist` dir.
**Network drive folders:** None
**Other details:** I haven't noticed too much in term of a pattern of when fuzzy search slows down.  Sometimes it works just fine and others it's essentially impossible (though I haven't let it run that long, never more than 20 seconds probably) to bring anything up with it besides recently opened files.  I think the only potential trigger I can identify is when I use the left-nav search function and that search goes extremely slowly.  That tends to correspond with a slowdown in the fuzzy search, but not always.

Happy to provide more information!  I love vscode, but if the fuzzy search continues to plague me I'll need to switch to something else :(.
